include/karabo-bridge/kb_client.hpp

```diff
@@ -629,11 +640,28 @@ class Client {
     }
 
 public:
-    Client(): ctx_(1), socket_(ctx_, ZMQ_REQ) {}
+    /*
+     * Constructor.
+     *
+     * @param timeout: connection timeout in milliseconds. "-1" (default) for infinite.
+     */
+    explicit Client(int timeout=-1): ctx_(1), socket_(ctx_, ZMQ_REQ) {
```

Comment:
Minor suggestion: the timeout parameter unit is `second` in the python client. We should maybe keep it the same here to stay consistent.

Reply:
Changed. But maybe we should have kept the default value in milliseconds at the beginning.

Reply:
I originally had it in milliseconds, Thomas and Hans argued for using seconds instead:
https://github.com/European-XFEL/karabo-bridge-py/pull/46#discussion_r239425916

Reply:
Thanks for the explanation! It's life.